File: application-web-mvc/src/main/java/com/osori/mvc/custom/AccountContext.java
package com.osori.mvc.custom;

import org.springframework.security.core.userdetails.UserDetails;

/**
 * 같은 쓰레드에서만 공유함
 */
public class AccountContext {
    public static final ThreadLocal<UserDetails> ACCOUNT_THREAD_LOCAL = new ThreadLocal<>();

    public static void setAccount(UserDetails account) {
        ACCOUNT_THREAD_LOCAL.set(account);
    }

    public static UserDetails getAccount() {
        return ACCOUNT_THREAD_LOCAL.get();
    }
}

File: application-web-mvc/src/main/kotlin/com/osori/mvc/custom/AccountContext.kt
package com.osori.mvc.custom

import org.springframework.security.core.userdetails.UserDetails

/**
 * 같은 쓰레드에서만 공유함
 */
class AccountContext {
    companion object {
        val ACCOUNT_THREAD_LOCAL = ThreadLocal<UserDetails>()
        fun setAccount (account: UserDetails) : Unit {
            ACCOUNT_THREAD_LOCAL.set(account)
        }

        fun getAccount () : UserDetails {
            return ACCOUNT_THREAD_LOCAL.get()
        }
    }
}